analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExceptionCandidate.cs

```diff
@@ -64,5 +70,14 @@ private ExceptionState ConversionExceptionCandidate(IOperationWrapperSonar opera
                        ? null
                        : new ExceptionState(typeCatalog.SystemInvalidCastException);
         }
+
+        private ExceptionState FromOperation(IMemberReferenceOperationWrapper reference) =>
+            reference.IsStaticOrThis() ? null : new ExceptionState(typeCatalog.SystemNullReferenceException);
+
+        private static ExceptionState FromOperation(IInvocationOperationWrapper invocation) =>
+            IsMonitorExit(invocation)
```

Comment:
I think we should add a comment here explaining the reasoning behind it. It's important to remember why this check is in the engine instead of providing an extension point for the rules.